miden-lib/src/transaction/events.rs

```diff
@@ -11,14 +11,22 @@ pub const EVENT_ID_PREFIX: u32 = 2;
 // TRANSACTION EVENT
 // ================================================================================================
 
-const ACCOUNT_VAULT_ADD_ASSET: u32 = 0x2_0000; // 131072
-const ACCOUNT_VAULT_REMOVE_ASSET: u32 = 0x2_0001; // 131073
-const ACCOUNT_STORAGE_SET_ITEM: u32 = 0x2_0002; // 131074
-const ACCOUNT_INCREMENT_NONCE: u32 = 0x2_0003; // 131075
-const ACCOUNT_PUSH_PROCEDURE_INDEX: u32 = 0x2_0004; // 131076
-const NOTE_CREATED: u32 = 0x2_0005; // 131077
-const ACCOUNT_STORAGE_SET_MAP_ITEM: u32 = 0x2_0006; // 131078
-const NOTE_ADD_ASSET: u32 = 0x2_0007; // 131079
+const ACCOUNT_VAULT_BEFORE_ADD_ASSET: u32 = 0x2_0000; // 131072
+const ACCOUNT_VAULT_AFTER_ADD_ASSET: u32 = 0x2_0001; // 131073
+
+const ACCOUNT_VAULT_BEFORE_REMOVE_ASSET: u32 = 0x2_0002; // 131074
+const ACCOUNT_VAULT_AFTER_REMOVE_ASSET: u32 = 0x2_0003; // 131075
+
+const ACCOUNT_STORAGE_BEFORE_SET_ITEM: u32 = 0x2_0004; // 131076
+const ACCOUNT_STORAGE_AFTER_SET_ITEM: u32 = 0x2_0005; // 131077
+
+const ACCOUNT_STORAGE_BEFORE_SET_MAP_ITEM: u32 = 0x2_0006; // 131078
+const ACCOUNT_STORAGE_AFTER_SET_MAP_ITEM: u32 = 0x2_0007; // 131079
+
+const ACCOUNT_INCREMENT_NONCE: u32 = 0x2_0008; // 131080
+const ACCOUNT_PUSH_PROCEDURE_INDEX: u32 = 0x2_0009; // 131081
+const NOTE_CREATED: u32 = 0x2_000A; // 131082
+const NOTE_ADD_ASSET: u32 = 0x2_000B; // 131083
```

Comment:
Similar comment to the one above. I would probably refactor these to have before/after events (except for `ACCOUNT_PUSH_PROCEDURE_INDEX` - this one may probably stay as is).